apps/devhub/tests/test_utils.py

```diff
@@ -0,0 +1,373 @@
+                         {'ignored': True,
+                          'matched': True})
+
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'ignore_duplicates': False,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': False,
+                          'matched': True})
+
+        # Other severities are ignored only when flagged as ignorable.
+        for severity in 'medium', 'high':
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': False,
+                          'matched': True})
+
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'ignore_duplicates': True,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': True,
+                          'matched': True})
+
+        # Check that unchecked properties don't matter.
+        self.compare({'id': ('a', 'b', 'c'),
+                      'signing_severity': 'low',
+                      'context': ('x', 'y', 'z'),
+                      'file': 'foo.js'},
+                     {'thing': 'stuff'},
+                     {'ignored': True,
+                      'matched': True})
+
+        # Non-signing messages should be matched, but not annotated for
+        # ignorability.
+        self.compare({'id': ('a', 'b', 'c'),
+                      'context': ('x', 'y', 'z'),
+                      'file': 'foo.js'},
+                     {},
+                     {'matched': True})
+
+    def test_non_matching_messages(self):
+        """Test that messages which should match, don't."""
+
+        message = {'id': ('a', 'b', 'c'),
+                   'signing_severity': 'low',
+                   'context': ('x', 'y', 'z'),
+                   'context_data': {'function': 'foo_bar'},
+                   'file': 'foo.js'}
+
+        self.compare(message, {'id': ('d', 'e', 'f')}, {})
+        self.compare(message, {'signing_severity': 'high'}, {})
+        self.compare(message, {'context': ('a', 'b', 'c')}, {})
+        self.compare(message, {'context_data': {}}, {})
+        self.compare(message, {'file': 'these-are-not-the-droids.js'}, {})
+
+        # Token non-signing message.
+        self.compare({'id': ('a', 'b', 'c'),
+                      'context': ('x', 'y', 'z'),
+                      'file': 'foo.js'},
+                     {'id': ()}, {})
+
+
+class TestValidationAnnotatorBase(amo.tests.TestCase):
+
+    def setUp(self):
+        # FIXME: Switch to factories.
```

Comment:
Fair enough. I decided to start writing `factory_boy` code for tests so I had a better idea of what the factories should look like before I started writing them, and I figured it would make sense to have the code already there in comments so I wouldn't have to write it again once the factories exist.

I did look at `*_factory`, but they didn't seem to actually be helpful here. And every time I see them used, I have to go back and read them to figure out what's going on, so I figured, better not.